.github/workflows/gradle.yml

```
@@ -57,7 +57,12 @@ jobs:
         cache: 'gradle'
 
     - name: Build with Gradle
-      run: ./gradlew clean build
+      run: |
+        latest_snapshot=$(curl -sf https://oss.sonatype.org/content/repositories/snapshots/org/seleniumhq/selenium/selenium-api/ | \
+          python -c "import sys,re; print(re.findall(r'\d+\.\d+\.\d+-SNAPSHOT', sys.stdin.read())[-1])")
+        echo ">>> $latest_snapshot"
+        echo "latest_snapshot=$latest_snapshot" >> "$GITHUB_ENV"
+        ./gradlew clean build -PisCI -Pselenium.version=$latest_snapshot
```

```suggestion
        ./gradlew clean build -PisCI -Pselenium.version=latest.integration
```
more details: https://docs.gradle.org/current/userguide/dependency_management.html#sec:dynamic_versions_and_changing_modules

however due to issues in publishing of Selenium snapshots (https://github.com/SeleniumHQ/selenium/issues/13106) it's needed to tune the resolution process in `build.gradle`:

```gradle
    if (project.hasProperty("isCI")) {
        maven {
            url uri('https://oss.sonatype.org/content/repositories/snapshots/')
            mavenContent {
                snapshotsOnly()
            }
            metadataSources {
                artifact()
            }
        }
    }
```

---

updated. I've also faced the above issue, just did not know whose bug it is

---

seems like it does not work: https://github.com/appium/java-client/actions/runs/6782494611/job/18434830083?pr=2067

---

that's true 😞  it seems there is no way to ignore `maven-metadata.xml`, but keep resolution of `*.pom`: all or nothing